docs: add Gainsight NXT connector documentation

JustinASmith opened the pull request:
**Description:**

- Documentation for the Gainsight NXT connector, detailing its capabilities, supported streams, prerequisites, configuration options, and sample usage.
- Update the README to include the Gainsight NXT connector in the list of available capture connectors.

**Workflow steps:**

(How does one use this feature, and how has it changed)

**Documentation links affected:**

(list any [documentation links](https://docs.google.com/document/d/1SRC9VS9zyCzWl3n4HXHbc4wPB1eLxJHkA2rtu9ZNokM/edit?usp=sharing) that you created, or existing ones that you've identified as needing updates, along with a brief description)

**Notes for reviewers:**

(anything that might help someone review this PR)

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/estuary/flow/2040)
<!-- Reviewable:end -->

Alex-Bair commented on site/docs/reference/Connectors/capture-connectors/README.md:
```diff
@@ -89,6 +89,9 @@ All Estuary connectors capture data in real time, as it appears in the source sy
 - Gladly
   - [Configuration](./gladly.md)
   - Package - ghcr.io/estuary/source-gladly:dev
+- Gainsight NXT
```
Could "Gainsight NXT" go before "Genesys" to keep the order alphabetical?

JustinASmith replied:
Whoops, yep I can get that updated!

Alex-Bair commented on site/docs/reference/Connectors/capture-connectors/gainsight-nxt.md:
```diff
@@ -0,0 +1,71 @@
+
+| Stream               | Capture Type | Cursor           |
+| -------------------- | ------------ | ---------------- |
+| `da_picklist`        | full-refresh | N/A              |
+| `companies`          | incremental  | ModifiedDate     |
+| `users`              | incremental  | ModifiedDate     |
+| `success_plans`      | incremental  | ModifiedDate     |
+| `cs_tasks`           | incremental  | ModifiedDate     |
+| `activity_timelines` | incremental  | LastModifiedDate |
+| `call_to_actions`    | incremental  | ModifiedDate     |
```
Are there links to relevant documentation on the endpoints we're hitting for each stream that could be hyperlinked here? Those are kind of nice to include; it's made it easier for me to get to the right docs without having to search through the code & the SaaS provider's website.

JustinASmith replied:
Yes, I can for most of these. However, success_plans technically isn't documented for how I am querying for it. I am using the V1 API for success_plan which uses the `cta_group` object name. This was to simplify development and not require supporting a new API and fetch flow just for 1 binding.

I could link the other API docs for that stream if we want that, but that might be confusing since we are not hitting that endpoint. Thoughts?

Alex-Bair commented on site/docs/reference/Connectors/capture-connectors/gainsight-nxt.md:
```diff
@@ -0,0 +1,71 @@
+
+| Property                         | Title             | Description                                                                                             | Type   | Required/Default                      |
+| -------------------------------- | ----------------- | ------------------------------------------------------------------------------------------------------- | ------ | ------------------------------------- |
+| `/credentials/access_token`      | API Access Key    | Gainsight API access key for authentication.                                                            | string | Required                              |
+| `/credentials/credentials_title` | Credentials Title | Name of the credentials set.                                                                            | string | Required, `"Private App Credentials"` |
+| `/domain`                        | Domain            | The domain for your Gainsight instance.                                                                 | string | Required                              |
+| `/start_date`                    | Start Date        | UTC date and time in the format 2021-01-25T00:00:00Z. Any data before this date will not be replicated. | string | Required, `30 day ago`                |
```
Typically, if a property is required in the endpoint or resource configs, the property name is bolded by enclosing it with `*`s.

Alex-Bair commented on site/docs/reference/Connectors/capture-connectors/gainsight-nxt.md:
```diff
@@ -0,0 +1,71 @@
+| Stream               | Capture Type | Cursor           |
+| -------------------- | ------------ | ---------------- |
+| `da_picklist`        | full-refresh | N/A              |
+| `companies`          | incremental  | ModifiedDate     |
+| `users`              | incremental  | ModifiedDate     |
+| `success_plans`      | incremental  | ModifiedDate     |
+| `cs_tasks`           | incremental  | ModifiedDate     |
+| `activity_timelines` | incremental  | LastModifiedDate |
+| `call_to_actions`    | incremental  | ModifiedDate     |
```
Super tiny nit: could the streams be sorted alphabetically/lexicographically at some level? The sort priority could be something like capture type -> cursor -> stream name.

Alex-Bair reviewed (approved):
LGTM % some small nits / conventions I've been following.

JustinASmith committed:
docs: add Gainsight NXT connector documentation

- Documentation for the Gainsight NXT connector, detailing its capabilities, supported streams, prerequisites, configuration options, and sample usage.
- Update the README to include the Gainsight NXT connector in the list of available capture connectors.
